### ISSUE TITLE
Add gradient descent with step decrease

### ISSUE THREAD

**Issue by ShkalikovOleh** [ShkalikovOleh/OptAlg#1]
Add gradient descent algorithm with step decrease via divide learning rate into fixed value if function value in new point greater than in previous point.
